lib/dynamic-files-router.js

```diff
@@ -0,0 +1,102 @@
+function DynamicRouter() {
+
+    var log = new Log("[dynamic-file-router]");
+
+    function getHandlebarsEnvironment(request, lookUpTable, configs) {
+        return (require("rendering-handlebars-helpers.js")).registerHelpers(request, lookUpTable, configs);
+    }
+
+    /**
+     * Returns a File object to the specified page.
+     * @param pageUri {string} page path requested in the URL
+     * @param pagesDirPath {string} path of the directory where pages are stored
+     * @return {File} requested page file
+     */
+    function getPageFile(pageUri, pagesDirPath) {
+        /*
+         * PAGE URI                REAL PATH
+         * /foo/bar         /app/pages/foo/bar.hbs OR /app/pages/foo/bar/index.hbs
+         * /foo/bar/        /app/pages/foo/bar.hbs OR /app/pages/foo/bar/index.hbs
```

Comment:
No longer valid because now pages are also shareable 'UI Component's and their _URI pattern_ is specified in their definition. See PR https://github.com/wso2/carbon-uuf/pull/2.